controller/action/AcquisitionActions.class.php

```diff
@@ -42,13 +42,73 @@ public static function executeYouTube(string $campaignId = '')
 
   public static function executeYT2()
   {
-    return ['acquisition/yt2', ['_no_layout' => true]];
+    return ['acquisition/yt2'];
   }
 
   public static function executeVerify(string $token)
   {
     return ['acquisition/verify', ['token' => $token]];
   }
 
+  public static function executeYoutubeToken()
+  {
+    return ['acquisition/youtube_token', ['_no_layout' => true]];
+  }
+
+  public static function executeYoutubeStatus(string $token)
+  {
+    return ['acquisition/youtube_status', ['token' => $token]];
+  }
+
+  public static function actionYoutubeToken(string $email, string $desired_lbry_channel_name, string $youtube_channel_id){
+
+    $email_is_valid = static::email_verification($email);
+    $desired_lbry_channel_name_is_valid = static::lbry_channel_verification($desired_lbry_channel_name);
+    $youtube_channel_id_is_valid = static::youtube_channel_verification($youtube_channel_id);
+
+
+    if ($email_is_valid && $desired_lbry_channel_name_is_valid && $youtube_channel_id_is_valid){
+      $token = LBRY::newYoutube($email, $youtube_channel_id, $desired_lbry_channel_name);
+
+      if ($token['error'] === null){
+        $url = "/youtube/status/" . $token['data']['status_token'];
+        Controller::redirect($url);
+      }
+      else{
+        $url = "/yt2?error=true&error_message=" . $token['error'];
+        Controller::redirect($url);
+      }
+    }
+    else{
+      $url = "/yt2?error=true";
+      Controller::redirect($url);
+    }
+  }
+
+  private static function email_verification($email){
```

Comment:
Unless you specifically need or want `private`, you should almost always use `protected` IMO.